test/e2etest.py

```diff
@@ -2636,6 +2637,52 @@ def test_twilio_delivery_update(fake_message_id):
     assert re.json()['twilio_delivery_status'] == 'delivered'
 
 
+def test_twilio_retry(fake_message_id):
+    if not fake_message_id:
+        pytest.skip('We do not have enough data in DB to do this test')
+
+    message_sid = uuid.uuid4().hex
+
+    with iris_ctl.db_from_config(sample_db_config) as (conn, cursor):
+        cursor.execute('''DELETE FROM `twilio_delivery_status` WHERE `message_id` = %s''', fake_message_id)
+        cursor.execute('''INSERT INTO `twilio_delivery_status` (`twilio_sid`, `message_id`)
+                          VALUES (%s, %s)''', (message_sid, fake_message_id))
+        cursor.execute('''DELETE FROM `twilio_retry` WHERE `retry_id` = %s OR `message_id` = %s''',
+                       (fake_message_id, fake_message_id))
+        conn.commit()
+
+    re = requests.post(base_url + 'twilio/deliveryupdate', data={'MessageSid': message_sid, 'MessageStatus': 'failed'})
+    assert re.status_code == 204
+
+    with iris_ctl.db_from_config(sample_db_config) as (conn, cursor):
+        cursor.execute('SELECT `message_id`, `retry_id` FROM `twilio_retry` WHERE `message_id` = %s', fake_message_id)
+        assert cursor.rowcount == 1
+        row = cursor.fetchone()
+        assert row[0] == fake_message_id
+        retry_id = row[1]
+        cursor.execute('SELECT * FROM `message` WHERE `id` = %s', retry_id)
```

Comment:
can we avoid `select *` ? if you just need to check for presence using .rowcount, you can do something like `select 1` or `select null`